sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala

```diff
@@ -93,8 +104,18 @@ class IntervalUtilsSuite extends SparkFunSuite {
     // Allow duplicated units and summarize their values
     checkFromString("1 day 10 day", new CalendarInterval(0, 11, 0))
     // Only the seconds units can have the fractional part
-    checkFromInvalidString("1.5 days", "Error parsing interval string")
-    checkFromInvalidString("1. hour", "Error parsing interval string")
+    checkFromInvalidString("1.5 days", "'days' cannot have fractional part")
+    checkFromInvalidString("1. hour", "'hour' cannot have fractional part")
```

Comment:
does `1. second` work?

Reply:
yes, it does. But I guess this should not work, we may just expose another bug in the new string to interval parser. we may need a follow-up to disable this.

Reply:
checked pgsql
```
cloud0fan=# select interval '1.' second;
ERROR:  invalid input syntax for type interval: "1."
LINE 1: select interval '1.' second;
```

we should fix it later.

Reply:
OK, I'll raise a jira and an pr later